azurelinuxagent/ga/exthandlers.py

```diff
@@ -949,14 +980,11 @@ def install(self):
         self.set_handler_state(ExtHandlerState.Installed)
 
     def uninstall(self):
-        try:
-            self.set_operation(WALAEventOperation.UnInstall)
-            man = self.load_manifest()
-            uninstall_cmd = man.get_uninstall_command()
-            self.logger.info("Uninstall extension [{0}]".format(uninstall_cmd))
-            self.launch_command(uninstall_cmd)
-        except ExtensionError as e:
```

Comment:
Now we'd only be swallowing exception if the user is trying to delete an extension but would not swallow it if the extension update fails.

Reply:
Do you mean the reverse? After your changes, a simple extension uninstall would result in an exception, but if it's part of an update sequence, we would swallow it and not block the update.

Reply:
No No this function `def uninstall(self):` is called twice - Once from `def handle_uninstall(self, ext_handler_i):` and other time from the `def _update_extension_handler(old_ext_handler_i, ext_handler_i):`. 
Initially we were swallowing exceptions inside this `def uninstall(self):` method which used to always eat the exception. I moved the swallow logic to this `def handle_uninstall(self, ext_handler_i):` method which is only called when the customer initiates an extension removal from the portal. Now the exception would not be swallowed during the update scenario where it will be caught and propagated as a `ExtensionUpdateError` if the `continueOnUpdateFailure` flag is not set.

Reply:
>  Now the exception would not be swallowed during the update scenario where it will be caught and propagated as a ExtensionUpdateError if the continueOnUpdateFailure flag is not set.

The current implementation doesn't distinguish between the new flag not being set, and it being set to false. We should change the behavior only when the flag is actually set (either to True or False)

Reply:
never mind, it's the same behavior